### PR TITLE
fix(preset-classic): throw if preset finds GA options in theme config

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Motivation

To prevent the legacy config from ever silently passing. Reported in https://github.com/facebook/docusaurus/pull/5832#issuecomment-1007122465.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Tested locally

### CODE REVIEW

**Comment by yangshun** [comment]
Thanks for this @Josh-Cena!

File: packages/docusaurus-plugin-google-analytics/src/index.ts [hunk 1]
```diff
@@ -82,7 +82,7 @@ export function validateOptions({
 export function validateThemeConfig({
   themeConfig,
 }: ThemeConfigValidationContext<ThemeConfig>): ValidationResult<ThemeConfig> {
-  if (themeConfig.googleAnalytics) {
```
Comment (slorber) [inline packages/docusaurus-plugin-google-analytics/src/index.ts:85]: I don't understand, why didn't this work before the change?

File: packages/docusaurus-preset-classic/src/index.ts [hunk 2]
```diff
@@ -48,6 +48,16 @@ export default function preset(
   if (algolia) {
     themes.push(require.resolve('@docusaurus/theme-search-algolia'));
   }
+  if ('gtag' in themeConfig) {
```
Comment (slorber) [inline packages/docusaurus-preset-classic/src/index.ts:51]: why is this addition needed?

Aren't plugins already throwing when running their own themeConfig validation? If it's not the case we have another deeper problem

Comment (slorber) [reply packages/docusaurus-preset-classic/src/index.ts:51]: > Edit. I see the reason now. If there's not a googleAnalytics in preset options, the plugin isn't even loaded and there's no theme config validation either

Oh I see, nasty 😅 Thanks for the fix

**Comment by yangshun** [comment]
I got bitten by this recently when upgrading my websites. My sites had much lower traffic for a few days before I realized why. Looking forward to the next version!

**Comment by slorber** [comment]
yes, we have a few features merged already, let's release this next week
